src/overlays/actors/ovl_Bg_Mori_Idomizu/z_bg_mori_idomizu.c

```diff
@@ -20,23 +35,152 @@ const ActorInit Bg_Mori_Idomizu_InitVars = {
+                     202);
+        return;
+    }
+    BgMoriIdomizu_SetupWaitForMoriTex(this);
+    sAlreadyLoaded = true;
+    this->isLoaded = true;
+    this->actor.room = -1;
+    // Forest Temple well water
+    osSyncPrintf("(森の神殿 井戸水)(arg_data 0x%04x)\n", this->actor.params);
+}
+
+void BgMoriIdomizu_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    s32 pad;
+    BgMoriIdomizu* this = THIS;
+
+    if (this->isLoaded) {
+        sAlreadyLoaded = false;
+    }
+}
+
+void BgMoriIdomizu_SetupWaitForMoriTex(BgMoriIdomizu* this) {
+    BgMoriIdomizu_SetupAction(this, BgMoriIdomizu_WaitForMoriTex);
+}
+
+void BgMoriIdomizu_WaitForMoriTex(BgMoriIdomizu* this, GlobalContext* globalCtx) {
+    if (Object_IsLoaded(&globalCtx->objectCtx, this->moriTexObjIndex)) {
+        BgMoriIdomizu_SetupMain(this);
+        this->actor.draw = BgMoriIdomizu_Draw;
+    }
+}
+
+void BgMoriIdomizu_SetupMain(BgMoriIdomizu* this) {
+    BgMoriIdomizu_SetupAction(this, BgMoriIdomizu_Main);
+}
+
+void BgMoriIdomizu_Main(BgMoriIdomizu* this, GlobalContext* globalCtx) {
+    s8 roomNum;
+    Actor* thisx = &this->actor;
+    s32 switchFlagSet;
+
+    roomNum = globalCtx->roomCtx.curRoom.num;
+    switchFlagSet = Flags_GetSwitch(globalCtx, thisx->params & 0x3F);
+    if (switchFlagSet) {
+        this->targetWaterLevel = -282.0f;
+    } else {
+        this->targetWaterLevel = 184.0f;
+    }
+    if (switchFlagSet && !this->prevSwitchFlagSet) {
+        func_800800F8(globalCtx, 0xCA8, 0x46, thisx, 0);
+        this->drainTimer = 90;
+    } else if (!switchFlagSet && this->prevSwitchFlagSet) {
+        func_800800F8(globalCtx, 0xCA8, 0x46, thisx, 0);
+        this->drainTimer = 90;
+        thisx->posRot.pos.y = 0.0f;
+    }
+    this->drainTimer--;
+    if ((roomNum == 7) || (roomNum == 8) || (roomNum == 9)) {
+        if (this->drainTimer < 70) {
+            Math_ApproxF(&thisx->posRot.pos.y, this->targetWaterLevel, 3.5f);
+            BgMoriIdomizu_SetWaterLevel(globalCtx, thisx->posRot.pos.y);
+            if (this->drainTimer > 0) {
+                if (switchFlagSet) {
+                    func_800788CC(NA_SE_EV_WATER_LEVEL_DOWN - SFX_FLAG);
+                } else {
+                    func_800788CC(NA_SE_EV_WATER_LEVEL_DOWN - SFX_FLAG);
+                }
+            }
+        }
+    } else {
+        thisx->posRot.pos.y = this->targetWaterLevel;
+        BgMoriIdomizu_SetWaterLevel(globalCtx, thisx->posRot.pos.y);
+        Actor_Kill(thisx);
+        return;
+    }
+    this->prevSwitchFlagSet = switchFlagSet;
+}
+
+void BgMoriIdomizu_Update(Actor* thisx, GlobalContext* globalCtx) {
+    s32 pad;
+    BgMoriIdomizu* this = THIS;
+
+    if (this->actionFunc != NULL) {
+        this->actionFunc(this, globalCtx);
+    }
+}
+
+void BgMoriIdomizu_Draw(Actor* thisx, GlobalContext* globalCtx) {
+    s32 pad;
+    BgMoriIdomizu* this = THIS;
+    u32 gFrames = globalCtx->gameplayFrames;
```

Comment:
this looks like the g from the s/g prefix for static and global. we can afford a longer name like gameplayFrames to be more explicit and avoid that confusion